c/meterpreter/source/extensions/stdapi/server/fs/search.c

```diff
@@ -839,9 +839,19 @@ DWORD request_fs_search(Remote * pRemote, Packet * pPacket)
 
 	if (!options.rootDirectory)
 	{
-		search_all_drives(&WDSInterface, &options, pResponse);
+		DWORD dwResult = ERROR_ACCESS_DENIED;
+		dwResult = search_all_drives(&WDSInterface, &options, pResponse);
+		if (dwResult != ERROR_SUCCESS)
+		{
+			dwResult = wds3_search(&WDSInterface, L"iehistory", NULL, &options, pResponse);
+		}
+		if (dwResult != ERROR_SUCCESS)
+		{
+			dwResult = wds3_search(&WDSInterface, L"mapi", NULL, &options, pResponse);
+		}
+		/*search_all_drives(&WDSInterface, &options, pResponse); 
 		wds3_search(&WDSInterface, L"iehistory", NULL, &options, pResponse);
-		wds3_search(&WDSInterface, L"mapi", NULL, &options, pResponse);
+		wds3_search(&WDSInterface, L"mapi", NULL, &options, pResponse);*/

```

Comment:
EDIT.... oh.... you were not the one that did this....  Still curious why it is here.......

Reply:
As you mentioned, I used the kind of construction based on the function "search". Actually, it is not necessary to specify variable dwResult, because subfunctions will change its' value further.